wordpress-mobile/WordPress-Android
# Implement Search for a domain screen

Fixes #19379

The PR introduces a brand-new search screen for domains that users would like to purchase or migrate. The PR is connected to the API and contains the entire search logic _except for error handling._
The search engine is similar to `DomainSuggestionsViewModel`: filters out free domains and sorts suggestions by relevance.
The `Transfer domain` button has UI implementation only so far.

To test:
- [x] Make sure you have enabled the `domain_management` feature flag.
- Go to Me -> Domains
- Tap on "+" button on the toolbar
- [x] Ensure you can see the `Search for domain` screen
- [x] Play with the search feature to confirm API calls work correctly

**Note:** I set the same debounce time as the one that already existed in `DomainSuggestionsViewModel` (250 ms) but I feel that it's not enough and causes unnecessary API calls while not extremely fast typing and it's worth having up to 300 ms instead.

https://github.com/wordpress-mobile/WordPress-Android/assets/16563318/fa806929-3af8-4daf-acf3-8d804c78819f

## Regression Notes
1. Potential unintended areas of impact
N/A.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing.

3. What automated tests I added (or what prevented me from doing so)
N/A.

PR submission checklist:


UI Changes testing checklist:
